File: implementation/scipamato/core/sync/src/test/java/ch/difty/scipamato/core/sync/jobs/code/CodeItemWriterTest.java
package ch.difty.scipamato.core.sync.jobs.code;

import static ch.difty.scipamato.common.TestUtilsKt.assertDegenerateSupplierParameter;

import org.jooq.DSLContext;
import org.junit.jupiter.api.Test;

import ch.difty.scipamato.core.sync.jobs.AbstractItemWriterTest;

class CodeItemWriterTest extends AbstractItemWriterTest<PublicCode, CodeItemWriter> {

    @Override
    protected CodeItemWriter newWriter(DSLContext dslContextMock) {
        return new CodeItemWriter(dslContextMock);
    }

    @Test
    void degenerateConstruction_withNullDslContext_throws() {
        assertDegenerateSupplierParameter(() -> new CodeItemWriter(null), "jooqDslContextPublic");
    }

}

File: implementation/scipamato/core/sync/src/test/kotlin/ch/difty/scipamato/core/sync/jobs/code/CodeItemWriterTest.kt
package ch.difty.scipamato.core.sync.jobs.code

import ch.difty.scipamato.core.sync.jobs.AbstractItemWriterTest
import org.jooq.DSLContext

internal class CodeItemWriterTest : AbstractItemWriterTest<PublicCode, CodeItemWriter>() {

    override fun newWriter(dslContextMock: DSLContext?) = CodeItemWriter(dslContextMock)

}
